### PR TITLE
feat(rust): Add rust-version to each Cargo.toml

### LINKED ISSUE #2204: Add `rust-version` to each `Cargo.toml`
[`rust-version` in Cargo.toml](https://github.com/rust-lang/rust/issues/65262) support landed in 1.56, so we should consider putting it in all of our Cargo.toml. I think 1.56 is also the version we support, so it would look like a `rust-version = "1.56.0"` line in them.

(If we actually support an older version, it doesn't break, but they get a warning)

### PR DESCRIPTION
**Pull request by Genysys**
## Current Behaviour
Documented in #2204 

## Proposed Changes

Add `rust-version` to all the `Cargo.toml`

N/A

This closes #2204 

## Checks

- [x] All commits in this Pull Request are [signed](https://docs.github.com/en/authentication/managing-commit-signature-verification/signing-commits).
- [x] All commits in this Pull Request follow the Ockam [commit message convention](https://www.ockam.io/learn/how-to-guides/contributing/CONTRIBUTING#commit-messages).
- [x] I accept the Ockam Community [Code of Conduct](https://www.ockam.io/learn/how-to-guides/high-performance-team/conduct/).
- [x] I have accepted the Ockam [Contributor Licence Agreement](https://www.ockam.io/learn/how-to-guides/contributing/cla/) by adding my Git/Github details in a row at the end of the [CONTRIBUTORS.csv](https://github.com/ockam-network/contributors/blob/master/CONTRIBUTORS.csv) file in a separate pull request to the [ockam-network/contributors](https://github.com/ockam-network/contributors) repository.

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -4,5 +4,6 @@
     "**/.pnp.*": true
   },
   "typescript.tsdk": "implementations/typescript/.yarn/sdks/typescript/lib",
-  "typescript.enablePromptUseWorkspaceTsdk": true
+  "typescript.enablePromptUseWorkspaceTsdk": true,
+  "java.configuration.updateBuildConfiguration": "interactive"
```
Comment (mrinalwadhwa) [inline .vscode/settings.json:8]: This line is an unrelated change, please remove it from this commit.

**Review by mrinalwadhwa — CHANGES_REQUESTED** [review]

**Comment by mrinalwadhwa** [comment]
@Genysys Welcome to the Ockam community and thank you for an awesome first PR!

I requested a small change above to the vscode code config that I think got accidentally included.
In addition to that, please also amend the commit message to:

```
feat(rust): improve formatting of `Cargo.toml`s  and add `rust-version` 1.56.0
```

File: .vscode/settings.json (hunk 1 above)
Comment (Genysys) [reply .vscode/settings.json:8]: taking it out , but it does break the json file

Comment (mrinalwadhwa) [reply .vscode/settings.json:8]: Hmm, interesting does vscode break for you without this change?

Comment (mrinalwadhwa) [reply .vscode/settings.json:8]: The json breaking may be because the previous line has an extra comma now, it should be removed too

Comment (Genysys) [reply .vscode/settings.json]: Silly vscode inserted the java config and i didnt realise it had changed. Removed now :)

**Comment by mrinalwadhwa** [comment]
@Genysys I just merged this. Thank you and congratulations on your first Ockam contribution 🎉
